packages/web/src/common/store/pages/collection/sagas.js

```diff
@@ -76,6 +86,42 @@ function* watchResetCollection() {
   })
 }
 
+/**
+ * Used for mobile CollectionScreen
+ */
+function* watchResetAndFetchCollectionTracks() {
+  yield takeEvery(
+    collectionActions.RESET_AND_FETCH_COLLECTION_TRACKS,
+    function* (action) {
+      const collectionUid = yield select(getCollectionUid)
+      const userUid = yield select(getUserUid)
+
+      if (collectionUid && userUid) {
+        // Reset collection so that lineup is not shared between separate instances
+        // of the CollectionScreen
+        yield put(collectionActions.resetCollection(collectionUid, userUid))
+        yield take(tracksActions.reset().type)
```

Comment:
why do we have to do this one differently than the others like `collectionActions.fetchCollectionSucceeded`

Reply:
This stumped me for a bit. I guess by default `take` calls the function if it's an action creator and uses the `type` value from the returned action. Well our lineup action creators use the `this` context to add a prefix, which was undefined when called by `take` implicitly

https://github.com/AudiusProject/audius-client/blob/main/packages/common/src/store/lineup/actions.ts#L208

Reply:
AHHHH cool. I remember seeing that on the lineup actions. Good to know!